samples/snippets/type_system_test.py

```diff
@@ -0,0 +1,359 @@
+
+    # Get the first character of each string
+    s.str[0]
+    # 0    a
+    # 1    d
+    # 2    1
+    # dtype: string
+
+    # Check whether there are only alphabetic characters in each string
+    s.str.isalpha()
+    # 0     True
+    # 1     True
+    # 2     False
+    # dtype: boolean
+
+    # Cast the alphabetic characters to their upper cases for each string
+    s.str.upper()
+    # 0    ABC
+    # 1     DE
+    # 2      1
+    # dtype: string
+    # [END bigquery_dataframes_type_system_string_accessor]
+    pandas.testing.assert_series_equal(
+        s.str[0].to_pandas(),
+        pd.Series(["a", "d", "1"], dtype=dtypes.STRING_DTYPE),
+        check_index_type=False,
+    )
+    pandas.testing.assert_series_equal(
+        s.str.isalpha().to_pandas(),
+        pd.Series([True, True, False], dtype=dtypes.BOOL_DTYPE),
+        check_index_type=False,
+    )
+    pandas.testing.assert_series_equal(
+        s.str.upper().to_pandas(),
+        pd.Series(["ABC", "DE", "1"], dtype=dtypes.STRING_DTYPE),
+        check_index_type=False,
+    )
+
+    # [START bigquery_dataframes_type_system_geo_accessor]
+    from shapely.geometry import Point
+
+    import bigframes.pandas as bpd
+
+    s = bpd.Series([Point(1, 0), Point(2, 1)])  # dtype: geometry
+
+    s.geo.y
+    # 0    0.0
+    # 1    1.0
+    # dtype: Float64
+    # [END bigquery_dataframes_type_system_geo_accessor]
+    pandas.testing.assert_series_equal(
+        s.geo.y.to_pandas(),
+        pd.Series([0.0, 1.0], dtype=dtypes.FLOAT_DTYPE),
+        check_index_type=False,
+    )
+
+    # [START bigquery_dataframes_type_system_json_query]
+    import db_dtypes
+    import pandas as pd
+    import pyarrow as pa
+
+    import bigframes.bigquery as bbq
+    import bigframes.pandas as bpd
+
+    fruits = [
+        '{"fruits": [{"name": "apple"}, {"name": "cherry"}]}',
+        '{"fruits": [{"name": "guava"}, {"name": "grapes"}]}',
+    ]
+
+    # db_dtypes.JSONArrowType() might be changed to pa.json_(pa.string()) in the future
+    json_s = bpd.Series(fruits, dtype=pd.ArrowDtype(db_dtypes.JSONArrowType()))
+    bbq.json_query(json_s, "$.fruits[0]")
+    # 0    {"name":"apple"}
+    # 1    {"name":"guava"}
+    # [END bigquery_dataframes_type_system_json_query]
+    pandas.testing.assert_series_equal(
+        bbq.json_query(json_s, "$.fruits[0]").to_pandas(),
+        pd.Series(['{"name":"apple"}', '{"name":"guava"}'], dtype=dtypes.JSON_DTYPE),
+        check_index_type=False,
+    )
+
+    # [START bigquery_dataframes_type_system_json_extract_array]
+    import db_dtypes
+    import pandas as pd
+    import pyarrow as pa
+
+    import bigframes.bigquery as bbq
+    import bigframes.pandas as bpd
+
+    fruits = [
+        '{"fruits": [{"name": "apple"}, {"name": "cherry"}]}',
+        '{"fruits": [{"name": "guava"}, {"name": "grapes"}]}',
+    ]
+
+    # db_dtypes.JSONArrowType() might be changed to pa.json_(pa.string()) in the future
```

Comment:
Same comment here re: being more explicit about what combination of pandas + pyarrow this will change.